### PR TITLE
LR enhancements for non-uniform payloads

### PR DESCRIPTION
**Pull request by marinhero**
This PR adds more enhancements and test coverage to the generateFile function. The most significant changes are:

- Headers Handling: We added a pass to gather all possible headers across the payloads at the start, ensuring the CSV file has consistent columns, even when some fields are missing in certain rows.

- Empty Field Support: When a payload is missing a particular field, the function now adds an empty value to keep the CSV structure intact. This prevents issues with misaligned rows in the output file.

- Added CSV fixtures and test coverage straight out of LR's website.

- The early header collection simplifies the row building thus improving the performance of the function overall.

## Testing

Testing completed successfully via expanded coverage of unit tests.

- [x] Added [unit tests](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing) for new functionality
- [ ] Tested end-to-end using the [local server](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing)
- [ ] [If destination is already live] Tested for backward compatibility of destination. **Note:** New required fields are a breaking change.
- [ ] [Segmenters] Tested in the staging environment
- [ ] [Segmenters] [If applicable for this change] Tested for regression with Hadron.

### CODE REVIEW

File: packages/destination-actions/src/destinations/liveramp-audiences/__tests__/__fixtures__/liveramp-good-example.csv
Comment (marinhero) [inline packages/destination-actions/src/destinations/liveramp-audiences/__tests__/__fixtures__/liveramp-good-example.csv:1]: This is just a sample file I'm preserving. Found it in LR's website. https://docs.liveramp.com/connect/en/file-formatting-examples.html#:~:text=The%20%22Under25%22%20field%20value%20is%20%22NULL%22%20rather%20than%20being%20left%20empty.

**Review by sayan-das-in — APPROVED** [review]

**Comment by marinhero** [comment]
Hey @joe-ayoub-segment thanks for merging this one in. Can you confirm it has been deployed? thanks!

**Comment by joe-ayoub-segment** [comment]
hi @marinhero yes it was deployed last week.

**Comment by joe-ayoub-segment** [comment]
hi @marinhero  PR deployed

**Pull request merged**